core/state/shared_pool.go

```diff
@@ -0,0 +1,85 @@
+package state
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"sync"
+)
+
+// sharedStorage is used to store maps of originStorage of stateObjects
+type SharedStorage struct {
+	poolLock   *sync.RWMutex
```

Comment:
No need use pointer for `sync.RWMutex`

Reply:
We need a consistent name. If you name the lock `poolLock`, who is the pool?  `SharedStorage` should be the pool? If yes, then `SharedStorage` is not a good name.